Task 7.1 leave group

Doenator310 opened the pull request:
Task 7.1 leave group completed

Doenator310 committed:
Started implementing basic leave functionality

Doenator310 committed:
Merge branch 'master' into task_7.1_leave_group

Doenator310 committed:
Implemented the ability to leave a group
Added an example page to leave the first group(database order) of the user logged in

Doenator310 committed:
Merge branch 'master' of https://github.com/DavidRisch/kicker into task_7.1_leave_group

Doenator310 committed:
Merged with master branch

Doenator310 committed:
Merge issue fixed

Doenator310 committed:
Task_7.1_leave_group completed

Doenator310 committed:
Update dump_db.js

Fixed an unnecessary whitespace

DavidRisch commented on api/api.js:
```diff
@@ -92,4 +97,7 @@ function init (app) {
     res.end(JSON.stringify(response))
   })
+
+  // ^^^ Insert new api calls here ^^^
```
Please remove

DavidRisch commented on api/leave_group.js:
```diff
@@ -0,0 +1,17 @@
+    // user not authenticated
+    return { success: false, error: 'userNotAuthenticated' }
+  }
+  // never trust user input!
```
No unnecessary humor

DavidRisch commented on api/leave_group.js:
```diff
@@ -0,0 +1,17 @@
+  if (require('../src/db/user').get_groups(user.id).filter(groupElement => groupElement.id === groupId).length > 0) {
+    require('../src/db/group').remove_User(groupId, user.id)
+  } else {
+    return { success: false, error: 'user cannot leave a group that he didn\'t joined!' }
```
Wrong grammar

DavidRisch commented on js/group_edit.js:
```diff
@@ -1,4 +1,5 @@
-/* global $ Dropzone */
+// this script requires jquery
+/* global apiPost, urlParams, $ Dropzone */
```
Unnecessary commas (this also applies to other files)

DavidRisch commented on page/group_edit.js:
```diff
@@ -1,8 +1,47 @@
 function page (req, res) {
+  const user = require('../src/account_util').get_current_user(req)
+  if (user === null) {
```
please use `require_logged_in_user` (applies to multiple files)

DavidRisch commented on page/page.js:
```diff
@@ -84,4 +88,6 @@ function init (app) {
     require('./faq').page(req, res)
   })
+
+  // ^^^ Insert new pages here ^^^
```
Please remove

DavidRisch commented on src/db/group.js:
```diff
@@ -114,6 +114,14 @@ function byName (name) {
 }
 
+function isGroupIdValid (id) {
+  if (isNaN(id)) { return 0 }
```
Seems wrong, please add an explanation (inline comment). Maybe use `Number.isInteger()`

DavidRisch commented on src/db/user_in_group.js:
```diff
@@ -0,0 +1,53 @@
+
```
This file was (rightfully) removed in 618b4c8b2996c8b54668e98e06591150ec931d17. Its functionality should be moved to db/group and/or db/user

DavidRisch reviewed (changes requested):
I have not yet tested the new features

Doenator310 committed:
Revised comments
Removed uneccessary user_in_group.js
Fixed grammar
Used "require_logged_in_user" instead of "get_current_user"

Doenator310 committed:
Merge branch 'task_7.1_leave_group' of https://github.com/DavidRisch/kicker into task_7.1_leave_group

DavidRisch committed:
Applied formatter

DavidRisch committed:
Fixed comment in db/group

DavidRisch reviewed (changes requested):
Looks good, but there are many linter warnings. Look at the other files for ways of fixing them
```
standard --fix
standard: Use JavaScript Standard Style (https://standardjs.com)
  /strg/notes/4_SoftwareEng/kicker/js/group_edit.js:2:19: 'urlParams' is defined but never used.
  /strg/notes/4_SoftwareEng/kicker/js/group_edit.js:67:3: 'alert' is not defined.
  /strg/notes/4_SoftwareEng/kicker/js/group_edit.js:71:3: 'alert' is not defined.
  /strg/notes/4_SoftwareEng/kicker/js/group_edit.js:75:3: 'alert' is not defined.
  /strg/notes/4_SoftwareEng/kicker/js/group_edit.js:81:12: 'myGroupInfo' is not defined.
  /strg/notes/4_SoftwareEng/kicker/js/group_edit.js:84:5: 'alert' is not defined.
  /strg/notes/4_SoftwareEng/kicker/js/group_edit.js:87:5: 'alert' is not defined.
  /strg/notes/4_SoftwareEng/kicker/js/group_edit.js:89:5: 'alert' is not defined.
  /strg/notes/4_SoftwareEng/kicker/js/group_edit.js:93:10: 'checkAndUpdateGroup' is defined but never used.
  /strg/notes/4_SoftwareEng/kicker/js/group_edit.js:94:7: 'myGroupInfo' is not defined.
  /strg/notes/4_SoftwareEng/kicker/js/group_edit.js:95:9: 'myGroupInfo' is not defined.
  /strg/notes/4_SoftwareEng/kicker/js/group_edit.js:106:10: 'abortGroupEdit' is defined but never used.
  /strg/notes/4_SoftwareEng/kicker/js/group_edit.js:110:10: 'deleteGroup' is defined but never used.
  /strg/notes/4_SoftwareEng/kicker/js/group_edit.js:114:10: 'leaveGroup' is defined but never used.
  /strg/notes/4_SoftwareEng/kicker/js/group_edit.js:115:7: 'myGroupInfo' is not defined.
  /strg/notes/4_SoftwareEng/kicker/js/group_edit.js:116:9: 'myGroupInfo' is not defined.
  /strg/notes/4_SoftwareEng/kicker/js/group_edit.js:128:7: 'myGroupInfo' is not defined.
  /strg/notes/4_SoftwareEng/kicker/js/group_edit.js:129:5: 'alert' is not defined.
  /strg/notes/4_SoftwareEng/kicker/js/group_edit.js:129:23: 'myGroupInfo' is not defined.
  /strg/notes/4_SoftwareEng/kicker/js/group_edit.js:132:30: 'myGroupInfo' is not defined.
  /strg/notes/4_SoftwareEng/kicker/js/group_edit.js:133:37: 'myGroupInfo' is not defined.
  /strg/notes/4_SoftwareEng/kicker/js/group_edit.js:135:38: 'myGroupInfo' is not defined.
  /strg/notes/4_SoftwareEng/kicker/js/group_edit.js:137:10: 'myGroupInfo' is not defined.
  /strg/notes/4_SoftwareEng/kicker/js/leave_group.js:2:19: 'urlParams' is defined but never used.
  /strg/notes/4_SoftwareEng/kicker/js/leave_group.js:8:16: 'onLeaveGroupBtnClicked' is defined but never used.
  /strg/notes/4_SoftwareEng/kicker/js/leave_group.js:11:12: 'myGroupInfo' is not defined.
  /strg/notes/4_SoftwareEng/kicker/js/leave_group.js:14:5: '$' is not defined.
  /strg/notes/4_SoftwareEng/kicker/js/leave_group.js:17:5: '$' is not defined.
  /strg/notes/4_SoftwareEng/kicker/js/leave_group.js:18:5: '$' is not defined.
  /strg/notes/4_SoftwareEng/kicker/js/leave_group.js:20:5: '$' is not defined.
  /strg/notes/4_SoftwareEng/kicker/js/leave_group.js:21:5: '$' is not defined.
  /strg/notes/4_SoftwareEng/kicker/js/leave_group.js:26:3: '$' is not defined.
  /strg/notes/4_SoftwareEng/kicker/js/leave_group.js:26:25: 'myGroupInfo' is not defined.
  /strg/notes/4_SoftwareEng/kicker/js/leave_group.js:27:7: 'myGroupInfo' is not defined.
  /strg/notes/4_SoftwareEng/kicker/js/leave_group.js:27:22: Expected '===' and instead saw '=='.
  /strg/notes/4_SoftwareEng/kicker/js/leave_group.js:28:5: '$' is not defined.
  /strg/notes/4_SoftwareEng/kicker/js/leave_group.js:30:5: '$' is not defined.
  /strg/notes/4_SoftwareEng/kicker/js/leave_group.js:34:1: '$' is not defined.
```

Doenator310 committed:
Fixed linter warnings
